es6/animations.js

```diff
@@ -5,180 +5,176 @@ import {
-    var {start_state, amt, end_state} = checked_animation_amt({start_state, end_state, amt, key})
-    var {start_time, end_time, duration} = checked_animation_duration({start_time, end_time, duration})
+const tick_func = ({duration, start_state, delta_state,
+                    curve='linear', unit=null}) => {
 
     const curve_func = EasingFunctions[curve]
 
-    return (delta) => {
-        let new_state
-        // These over-boundary cases happen because we need to render
-        // an extra frame before/after start/end times to finish off
-        // animations who's (durations) % (frame rate) != 0.
-        if (delta < 0) {
-            new_state = start_state
-        } else if (delta >= duration) {
-            new_state = end_state
-        } else {
-            // tick progression function, the core math at the heart of animation
-            new_state = start_state + curve_func(delta/duration)*amt
-        }
-        return unit ? `${new_state}${unit}` : new_state
+    return (time_elapsed) => {
+        const bounded_time = Math.min(Math.max(time_elapsed, 0), duration)
+        const curve = curve_func(bounded_time/duration)
+        const new_state = start_state + curve * delta_state
+
+        return add_unit(new_state, unit)
     }
 }
 
-const checked_animation_duration = ({start_time, duration, end_time}) => {
-    if ([start_time, end_time, duration].filter(a => typeof(a) == 'number').length < 2) {
-        console.log({start_time, end_time, duration})
-        throw 'Need at least 2/3 to calculate animation: start_time, end_time, duration'
-    }
+const add_unit = (val, unit) => {
+    return unit ? `${val}${unit}` : val
+}
 
-    if (start_time === undefined)
-        start_time = end_time - duration
-    if (end_time === undefined)
-        end_time = start_time + duration
-    if (duration === undefined)
-        duration = end_time - start_time
+export const Become = ({path, state, start_time,
+                        end_time=Infinity, duration=Infinity}) => {
+    if (start_time === undefined) start_time = Date.now()
 
-    if (start_time + duration != end_time) {
-        console.log({start_time, end_time, duration})
-        throw 'Conflicting values, Animation end_time != start_time + duration'
+    if (exactlyOneIsUndefined(duration, end_time)) {
+        [duration,
+             end_time] = computeTheOther(start_time, duration, end_time)
+    }
+    if (start_time === undefined || path === undefined) {
+        // console.log({path, state, start_time, end_time, duration})
+        throw 'Become animation must have a start_time and path defined.'
     }
-    return {start_time, duration, end_time}
+    return Animate({
+        type: 'BECOME',
+        path,
+        start_state: state,
+        end_state: state,
+        start_time,
+        end_time,
+        duration,
+        tick: (time_elapsed) => {
```

Comment:
`tick: (_) => state,` is clearer because it's explicit that elapsed_time doesn't matter for become.  Also this code breaks eslint for unused-vars.